scripts/trigger_pipeline.sh

```diff
@@ -0,0 +1,205 @@
+      t | trigger-token )       needs_arg ; CI_JOB_TOKEN="${OPTARG}" ;;
+      a | access-token )        needs_arg ; PIPELINE_TOKEN="${OPTARG}" ;;
+      ??* )                     log DIE "Illegal option --${OPT}" ;;  # bad long option
+      ? )                       exit 2 ;;  # bad short option (error reported via getopts)
+    esac
+  done
+  shift $((OPTIND-1)) # remove parsed options and args from $@ list
+
+}
+
+function check_args {
+  if [[ -z "${SIMNET_VERSION}" ]] ; then
+    log DIE "Must specify value for mandatory argument -s,--simnet-version
+
+$(usage)"
+  fi
+}
+
+function needs_arg {
+  if [ -z "${OPTARG}" ]; then
+    log DIE "No arg for --${OPT} option"
+  fi
+}
+
+function trigger_pipeline {
+  # API trigger another project's pipeline.
+  log INFO "Triggering Simnet pipeline."
+
+  curl --silent \
+      -X POST \
+      -F "token=${CI_JOB_TOKEN}" \
+      -F "ref=${SIMNET_VERSION}" \
+      -F "variables[TRGR_PROJECT]=${TRGR_PROJECT}" \
+      -F "variables[TRGR_REF]=${TRGR_REF}" \
+      -F "variables[IMAGE_NAME]=${IMAGE_NAME}" \
+      -F "variables[IMAGE_TAG]=${IMAGE_TAG}" \
+      -F "variables[FEATURES]=${FEATURES}" \
+      "https://${CI_SERVER_HOST}/api/v4/projects/${DWNSTRM_ID}/trigger/pipeline" | \
+          tee pipeline;
+}
+
+function check_pipeline {
+  PIPELINE_ID=$(jq ".id" pipeline)
+  PIPELINE_URL=$(jq ".web_url" pipeline)
+  echo
+  log INFO "Simnet pipeline ${PIPELINE_URL} was successfully triggered."
+  log INFO "Now we're polling it to obtain the distinguished status."
+}
+
+function poll_pipeline {
+  # This is a workaround for a Gitlab bug, waits here until
+  # https://gitlab.com/gitlab-org/gitlab/-/issues/326137 gets fixed.
+  # The timeout is 360 curls with 8 sec interval, roughly an hour.
+  log INFO "Waiting on ${PIPELINE_ID} status..."
+
+# shellcheck disable=SC2034
+  for i in {1..360}; do
+      STATUS=$(get_status);
+      log INFO "Triggered pipeline status is ${STATUS}";
+      if [[ ${STATUS} =~ ^(pending|running|created)$ ]]; then
+          echo;
+      elif [[ ${STATUS} =~ ^(failed|canceled|skipped|manual)$ ]]; then
+          log DIE "Something's broken in: ${PIPELINE_URL}";
+      elif [[ ${STATUS} =~ ^(success)$ ]]; then
+          log INFO "Look how green it is: ${PIPELINE_URL}"
+          exit 0
+      else
+          log DIE "Something else has happened in ${PIPELINE_URL}"
+      fi
+  sleep 8;
+  done
+}
+
+function get_status() {
+    curl --silent \
+        --header "PRIVATE-TOKEN: ${PIPELINE_TOKEN}" \
+        "https://${CI_SERVER_HOST}/api/v4/projects/${DWNSTRM_ID}/pipelines/${PIPELINE_ID}" | \
+            jq --raw-output ".status";
+}
+
+function log {
+  local lvl msg fmt
+  lvl=$1 msg=$2
+  fmt='+%Y-%m-%d %H:%M:%S'
+  lg_date=$(date "${fmt}")
+  if [[ "${lvl}" = "DIE" ]] ; then
+    lvl="ERROR"
+    echo  "${lg_date} - ${lvl} - ${msg}"
+    exit 1
+  else
+    echo "${lg_date} - ${lvl} - ${msg}"
+  fi
+}
+
+main "$@"
```

Comment:
is there any difference between 3 `trigger_pipeline.sh` files in substrate, polkadot and here?

Reply:
please mention it in the comment above the difference, or you'll forget about it soon

Reply:
why did you leave this script here?